specification/dartLangSpec.tex

```diff
@@ -16023,8 +16042,33 @@ \subsection{Await Expressions}
 (\ref{expressionEvaluation}).
 If $f$ completes with an object $v$, $a$ evaluates to $v$.
 
-% Otherwise, the value of $a$ is the value of $e$. If evaluation of
-% $e$ raises an exception $x$, $a$ raises $x$.
+\rationale{%
+This use of \flattenName{} implies that we await a future
+in every case where this choice is sound.%
```

Comment:
Not clear which use of _flatten_ this refers to.

It's actually the use of the type `T`, derived from _flatten_, in the "is a subtype of `Future<T>`" which implies that we await as much as possible.

So, maybe:

> Using `Future<T>` as the check for which futures to await implies that
> we await every future that we can soundly await.

Or add it below in the commentary instead.

Reply:
I think this is clearer now.